File: 2025/09/main.py
from utils import *


def area(a, b):
    return (abs(a[0] - b[0]) + 1) * (abs(a[1] - b[1]) + 1)


#########################################################
#########################################################
@performance
def solve(__file, pt):
    lines = [line.strip() for line in open(__file)]
    if len(lines) == 0:
        return -1

    res = 0
    reds = []
    for i, line in enumerate(lines):
        c, r = ints(line)
        reds.append((r, c))

    for i in range(len(reds) - 1):
        for j in range(i + 1, len(reds)):
            res = max(res, area(reds[i], reds[j]))
    return res


#########################################################
#########################################################

if __name__ == "__main__":
    input_file = sys.argv[1] if len(sys.argv) > 1 else "input.txt"
    i_sol1 = solve(input_file, 1)
    print(f"\033[1m\033[92m[SOLUTION] {i_sol1 = }\033[0m")
    # i_sol2 = solve(input_file, 2)
    # print(f"\033[1m\033[92m[SOLUTION] {i_sol2 = }\033[0m")

File: 2025/09/main.go
package main

import (
	"fmt"
	"os"

	"aoc/utils"
)

func solve(filename string, pt int) int {
	lines, err := utils.ReadLines(filename)
	if err != nil || len(lines) == 0 {
		return -1
	}

	res := 0
	for i, line := range lines {
		_ = i
		_ = line
	}

	return res
}

func main() {
	inputFile := "input.txt"
	if len(os.Args) > 1 {
		inputFile = os.Args[1]
	}

	sol1 := utils.Performance("solve1", func() int { return solve(inputFile, 1) })
	fmt.Printf("\033[1m\033[92m[SOLUTION] i_sol1 = %d\033[0m\n", sol1)

	sol2 := utils.Performance("solve2", func() int { return solve(inputFile, 2) })
	fmt.Printf("\033[1m\033[92m[SOLUTION] i_sol2 = %d\033[0m\n", sol2)
}
